src/gpu/multiexp.rs

```diff
@@ -165,16 +171,22 @@ impl<E> MultiexpKernel<E> where E: Engine {
         }
 
         let num_devices = self.0.len();
-        let chunk_size = ((n as f64) / (num_devices as f64)).ceil() as usize;
 
         // Bases are skipped by `self.1` elements, when converted from (Arc<Vec<G>>, usize) to Source
         // https://github.com/zkcrypto/bellman/blob/10c5010fd9c2ca69442dc9775ea271e286e776d8/src/multiexp.rs#L38
         let bases = &bases[skip..];
 
         let exps = &exps[..n];
 
+        let cpu_n = ((n as f64) / ((num_devices as f64) * SPEEDUP + 1.0f64)).ceil() as usize;
```

Comment:
could you add some comments where this formula comes from/what the thinking behind it is?